Removes most of the enforced name rules

Recoherent opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Guts most of the reject_bad_name proc to allow a larger degree of freedom when naming characters. This includes removal of automatic-capitalization and allows all species to use punctuation and numbers in their names. Sanitization of spaces still remains; this won't allow for massive blocks of spaces or leading/trailing whitespace or anything.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Ever got pissed that you couldn't put a number in your plasmaman's name, or that you couldn't name an agent card after your IPC?
This pretty obviously makes the name system more abusable, but that's the sort of thing that's easy to handle with admin intervention. I hold that the benefits of more open naming far outweigh opening it up to extremely obvious shitter behavior.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
tweak: Name enforcement is more open now; put those numbers on your agent card or your bizarre-super-soldier-clone-test-project-human's name!
code: Removed reference to allow_numbers from saved preferences.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Recoherent committed:
Moves ash/salt garnish to crafting menu

Fixes bonfire crashes and declutters weird chemistry for these two. wowowowowoowowowowowwwwowowoww

Recoherent committed:
Merge branch 'master' of https://github.com/shiptest-ss13/Shiptest

Recoherent committed:
Guts name sanitization

Gottem, now you can do whatever the hell you want within ascii reason.

Calyxman commented:
'bizarre-super-soldier-clone-test-project-human's name!'
Wow you didn't have to be so direct

Fikou commented:
> I hold that the benefits of more open naming far outweigh opening it up to extremely obvious shitter behavior.

I disagree, if you want to you can let agent cards use numbers, but I am not interested in this.

Calyxman commented:
> > I hold that the benefits of more open naming far outweigh opening it up to extremely obvious shitter behavior.
> 
> I disagree, if you want to you can let agent cards use numbers, but I am not interested in this.

why can't the admins just manually enforce it?

PrefabQuasar commented:
> > I hold that the benefits of more open naming far outweigh opening it up to extremely obvious shitter behavior.
> 
> I disagree, if you want to you can let agent cards use numbers, but I am not interested in this.

I see no reason to disallow human characters from using numbers in their names, which is essentially all this PR changes, you could already name yourself **incredibly** dumb things, and all things considered, its easier to police bad names than it is to have shitters prevented from outing themselves lurking on the server.

tiramisuapimancer commented:
as possibly the lore head nobodys sure. this is based and cool actually

Fikou commented:
> I see no reason to disallow human characters from using numbers in their names, which is essentially all this PR changes

that's not entirely right though, it lets everything no longer be capitalized and lets you put all base ascii in your name, this applies to a few more things than just character names as well.
i see no reason to remove functionality from the generic proc, when you can just use args for this.

i am also against unlimiting it in general but whatever, if the lore people want to let it for character names sure.

ZephyrTFA commented:
I'm blocking this as a server issue due to the fact that names are almost never sanitized properly. The changes you have made to the name check proc allows html code through.
Until this is resolved I'm blocking this as a server breaking issue

Recoherent commented:
Got it, didn't consider sanitization, thanks for bringing that up. I'll pull back to the previously allowed set of characters but keep the relaxed rules.

Recoherent committed:
no more arbitrary html code! *vine thud sfx*

thanks zeph, this would have been real bad to let through. let me know if any other characters would be safe to add

Fikou commented on code/modules/client/preferences_savefile.dm:
```diff
@@ -461,14 +461,13 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 
 	//Sanitize
-	real_name = reject_bad_name(real_name, pref_species.allow_numbers_in_name)      // WS Edit - numbers in IPC names load from save
+	real_name = reject_bad_name(real_name)      // WS Edit - numbers in IPC names load from save
```
remove the comment

Fikou reviewed (dismissed):
This will make renaming robot parts limited to TRUE length (or 1), fix that

Fikou commented on code/modules/client/preferences.dm:
```diff
@@ -2404,7 +2404,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
-		var/sanitized_name = reject_bad_name(raw_name,namedata["allow_numbers"])
+		var/sanitized_name = reject_bad_name(raw_name)
 		if(!sanitized_name)
 			to_chat(user, "<font color='red'>Invalid name. Your name should be at least 2 and at most [MAX_NAME_LEN] characters long. It may only contain the characters A-Z, a-z,[namedata["allow_numbers"] ? ",0-9," : ""] -, ' and .</font>")
```
fix this.

Recoherent replied:
for clarification, is this just for fixing the feedback message?

ZephyrTFA reviewed (approved):
The issues I've indicated are resolved, however you still need to address Fikou's review

Recoherent committed:
cleanup, robot parts rename right, name feedback actually right

wwwwwwwwwwwwwwwwwwwww thanks fikou

Recoherent commented:
their review was addressed, just need them to sign off on it I suppose? just waiting until they're back
